ocaml-lsp-server/src/client.ml

```diff
@@ -7,17 +7,16 @@ module Vscode = struct
         ~command:"editor.action.triggerSuggest" ()
 
     module Custom = struct
-      let next_hole ?start_position ~notify_if_no_hole () =
+      let next_hole ?in_range ~notify_if_no_hole () =
         let arguments =
           let arg_obj_fields =
             let notif_json =
-              Some ("notify-if-no-hole", Json.bool notify_if_no_hole)
+              Some ("shouldNotifyIfNoHole", Json.bool notify_if_no_hole)
```

Comment:
Isn't this a breaking change? What about old clients?

Reply:
Ah, indeed, that's a breaking change. When working on the PR, I thought we'd be releasing one more ocaml-lsp for 4.12, which would have the fix. I can provide backward-compatibility or we could have one more release (I can handle it if you prefer). With backward compat, users will have everything working but if they happen to have a typed hole outside the newly generated text and the newly generated text doesn't, the cursor may jump away... (sorry)

```ocaml
let a = _ (* hole here, where the jump happens *)

type r = { k : int } 
let f ( r : r ) = (* when the user destructs the value [r], they get `{ k }`, so the cursor jumps to the first line *)
  ()
```

Reply:
I can preserve the current structure `{ "notify-if-no-holes": <bool>; "position": <Position.t> }` and add a field `end-position: <Position.t>` to preserve somewhat backward compat. It'll result in an ugly solution, I'm afraid. Sorry for the mess. Previously, when we introduced custom requests we never had such problems (because they had simpler param requirements), and I haven't thought about making the params more flexible for future improvements 🙏

Reply:
If it's too much work to maintain backwards compatibility, we don't need to bother. At the end of the day, this feature is a small part of the plugin and users can live without it until they upgrade. But at the very least we should:

* Mention the breakage somewhere.
* Make the error message explicit about the breakage. So the old method should now at least say that this "method is renamed xx and is no longer supported".

I think in this case maintaining support for the old method is easy enough.

Reply:
This issue still stands.

Reply:
I made the extension to drop command invocations with old-style arguments (that change is in the counterpart PR in the extension repo). 

Since the extension with the command supporting old-style arguments was never released, users who use the older ocaml-lsp (ocaml < 4.13) will simply never know this feature (jumping to a hole on destruct/construct) existed, so they don't need to be worried or notified about anything.

People on a newer ocaml-lsp will enjoy the feature as if nothing happened. 

Does this make sense?

Reply:
Okay. If it was never released, then there's no problem.